beacon-chain/blockchain/process_block.go

```diff
@@ -212,15 +212,17 @@ func (s *Service) onBlock(ctx context.Context, signed interfaces.SignedBeaconBlo
 
 	// Save justified check point to db.
 	if justified.Epoch > currJustifiedEpoch {
-		if err := s.cfg.BeaconDB.SaveJustifiedCheckpoint(ctx, postState.CurrentJustifiedCheckpoint()); err != nil {
+		if err := s.cfg.BeaconDB.SaveJustifiedCheckpoint(ctx, &ethpb.Checkpoint{
+			Epoch: justified.Epoch, Root: justified.Root[:],
+		}); err != nil {
```

Comment:
Was this wrong before or saving copies?

Reply:
A block may result in justifying a different one.  In this case forkchoice will return a different justified checkpoint than the post-state. In any case I feel safer saving forkchoice's justified checkpoint since this is the one that we will be acting upon, instead of this particular block's postate.